### PR TITLE
feat: enable vote casting form again on offchain active proposal

### PR DESCRIPTION
**Pull request by wa0x6e**
### Summary

Closes: #513

Allow vote recasting on active offchain proposals.

 
#### On onchain proposals

- Voting flow for unvoted proposals remain the same 
- On voted proposals, it should show your choice in the sidebar 

Screenshot 2024-07-23 at 12 37 31

#### On offchain proposals

1. Go to an offchain proposal you have not voted on
2. It should show the cast vote form
3. Sign and submit a vote
4. It should show your choice, with a edit icon

Screenshot 2024-07-23 at 12 39 26

5. Clicking on the button should show the vote form again, pre-populated with your previous vote choices

Screenshot 2024-07-23 at 12 39 44

On proposals you have already voted on, it should show the button on step 4

On shielded proposals, it should not show any choices, and the vote form is not pre-populated with any choices 

Screenshot 2024-07-23 at 12 41 50

### CODE REVIEW

File: apps/ui/src/helpers/utils.ts
Comment (wa0x6e) [reply apps/ui/src/helpers/utils.ts:481]: blank vote is possible, I tested with Sam, and we handled that on snapshot

Comment (ChaituVR) [reply apps/ui/src/helpers/utils.ts:481]: is there some example vote?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: apps/ui/src/components/ProposalVote.vue
Comment (wa0x6e) [reply apps/ui/src/components/ProposalVote.vue]: Seems like there's some eslint rule for that.

Should we add it ? https://eslint.vuejs.org/rules/custom-event-name-casing

**Review by ChaituVR — APPROVED** [review]
tAck

File: apps/ui/src/components/ProposalVote.vue
Comment (Sekhmet) [reply apps/ui/src/components/ProposalVote.vue]: We don't have many past examples for casing for events (pretty much all we have is single-word event names), `getVotingPower` is the exception.

I'd say here we should match camelCase and add eslint rule for this for whatever we end up picking (I'd say `camelCase` is more natural).

File: apps/ui/src/composables/useAccount.ts
Comment (Sekhmet) [inline apps/ui/src/composables/useAccount.ts:38]: How does it affect [this](https://github.com/snapshot-labs/sx-monorepo/pull/518#discussion_r1687597637)?

**Review by Sekhmet — CHANGES_REQUESTED** [review]
Block this PR as it already has approval so we don't merge it by accident.

File: apps/ui/src/components/ProposalVote.vue
Comment (ChaituVR) [reply apps/ui/src/components/ProposalVote.vue]: found this one. and works if i add a rule

```JSON
"rules": {
  "vue/custom-event-name-casing": ["error", "camelCase"]
}
```

**Review by bonustrack — APPROVED** [review]
tACK

File: apps/ui/src/composables/useAccount.ts
Comment (wa0x6e) [reply apps/ui/src/composables/useAccount.ts]: Reverted, this fix was extracted and improved in #518

**Review by Sekhmet — COMMENTED** [review]
Right now If I vote on proposal for the first time I will get quick flash of "You already voted for this proposal" and only then I will get edit option.

Maybe we should do something like this to avoid it?
```diff
diff --git a/apps/ui/src/components/ProposalVote.vue b/apps/ui/src/components/ProposalVote.vue
index 2d432724..3792beb6 100644
--- a/apps/ui/src/components/ProposalVote.vue
+++ b/apps/ui/src/components/ProposalVote.vue
@@ -81,7 +81,11 @@ const isEditable = computed(() => {
     </div>
   </slot>
   <slot
-    v-else-if="!isEditable && pendingVotes[proposal.id]"
+    v-else-if="
+      !isEditable &&
+      pendingVotes[proposal.id] &&
+      !offchainNetworks.includes(props.proposal.network)
+    "
     name="voted-pending"
   >
     You have already voted for this proposal
```

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**
